kyo-scheduler/jvm/src/main/scala/kyo/scheduler/util/LoomSupport.scala

```diff
@@ -0,0 +1,40 @@
+package kyo.scheduler.util
+
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+import java.util.concurrent.ThreadFactory
+import java.util.logging.Logger
+import scala.util.control.NonFatal
+
+object LoomSupport:
+
+    private val log = Logger.getLogger(getClass.getName())
+
+    def tryVirtualize(enabled: Boolean, exec: Executor): Executor =
+        if !enabled then exec
+        else
+            try
+                val ofVirtual = classOf[java.lang.Thread].getMethod("ofVirtual").invoke(null)
+
+                val schedulerField = ofVirtual.getClass.getDeclaredField("scheduler")
+                schedulerField.setAccessible(true)
+                schedulerField.set(ofVirtual, exec)
+
+                val factoryMethod = ofVirtual.getClass.getDeclaredMethod("factory")
+                factoryMethod.setAccessible(true)
+                val factory = factoryMethod.invoke(ofVirtual)
+
+                classOf[Executors]
+                    .getMethod("newThreadPerTaskExecutor", classOf[ThreadFactory])
+                    .invoke(null, factory).asInstanceOf[Executor]
```

Comment:
Cool, I just did the same too
 toohttps://github.com/apache/rocketmq/pull/8063

Reply:
So you don't plan to pooling the virtual thread right?

Reply:
Cool! I tried polling but putting virtual threads to sleep is too expensive and it's cheaper to allocate new ones. Virtual threads are allocated only when workers start, though. A worker mounted on a virtual thread can several tasks until there's no more pending work, which alleviates the cost of the virtual thread creation on each task execution.

Reply:
Yeah,  If totally move to none pooling will need to change a lot of the current work base at work, where thread pool is using like some kind of concurrency limiter too